### PR TITLE
Added clashing events detection and relevant warning messages

### LINKED ISSUE #95: Alert Users on Clashing Schedules
Alerts users if there is a clash in their timetable.

### LINKED ISSUE #169: [PE-D] Able to add the same event at the same time
I am able to add the same event at the same time.

Maybe check for similar events so users who forgot they added an event wouldn't add the event again.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: 0xZ3RO/ped#4

### PR DESCRIPTION
**Pull request by brandonywl**
Closes #169. Closes #95.

Clash detection works by checking if the added event or edit event's resulting event has the a clashing time period, and they have the same title.

If they have the same title, a specific message on duplication warning will show. If not just a normal clash message shows.

### CODE REVIEW

**Review by Chongjx — APPROVED** [review]
LGTM

**Review by R-Ramana — APPROVED** [review]

File: src/main/java/seedu/notus/data/timetable/Event.java [hunk 1]
```diff
@@ -231,4 +287,15 @@ public String toString() {
     public int compareTo(Event o) {
         return startDateTime.compareTo(o.startDateTime);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o instanceof Event) {
+            Event event = (Event) o;
+            if (compareTo(event) == 0 && title.equals(event.title)) {
```
[the remark this answers is not in the document]
Comment (brandonywl) [reply src/main/java/seedu/notus/data/timetable/Event.java:295]: Noted will do.

**Comment by brandonywl** [comment]
Waiting for PR #189 to merge first with all changes approved before this will merge.

**Comment by brandonywl** [comment]
Similarity detection was change from only same start time to when there is a clash in timing.

**Pull request merged**